### ISSUE TITLE
Getting stuck on "adding middlewares"

### ISSUE THREAD

**Issue by eneroth** [tonsky/Clojure-Sublimed#43]
I'm attempting to connect to a server hosted by Shadow-CLJS:

```clojure 
{:deps        {:aliases [:common]}
 :nrepl       {:port       5555
               :middleware []}
 …}
```

There are no error messages in the log, as I can see.

![Screenshot 2021-12-14 at 10 54 57](https://user-images.githubusercontent.com/333974/145975460-022892e9-2f07-4248-9377-ee74bf2abea9.png)

**Comment by jaihindhreddy** [comment]
Please make sure you are using `nrepl` version `0.8` or later.
I faced this issue when I used a very old version of `nrepl`.

**Comment by eneroth** [comment]
Ah, I have no idea which REPL is included in Shadow-CLJS. It might be an old one, I guess.

**Comment by tonsky** [comment]
I have never tested with CLJS. There are a few middlewares which are clojure-native, not sure how it goes with CLJS REPL. If you can figure out what’s going on and what’s needed to make CLJS tick, it would be a great help. See also #31

**Comment by eneroth** [comment]
Shadow-CLJS exposes a normal CLJ socket REPL as well as an NREPL, not a CLJS one. Once connected to the CLJ REPL, it's possible to tunnel into the CLJS one. But I have no idea how to connect to a CLJS REPL directly.

**Comment by sepisoad** [comment]
I'm also suffering from the same issue

**Comment by alekzvik** [comment]
**TL;DR** shadow-cljs implements nrepl protocol for v0.4. Clojure Sublimed expects at least 0.8. 

I encountered the same issue. After I turned the `debug: True`, I noticed that shadow-cljs's nrepl replies with `unknown-op`

A few hours later, I found out that shadow-cljs redefines middlewares for nrepl and [requires them one by one here](https://github.com/thheller/shadow-cljs/blob/051b5ecc7d701ef45b3bbc839685ac6ea03d4ad7/src/main/shadow/cljs/devtools/server/nrepl.clj#L104:L113)
This piece of code was introduced for nrepl version 0.4, and after comparing to [the list in v1.0.0](https://github.com/nrepl/nrepl/blob/master/src/clojure/nrepl/server.clj#L96:L107), I found four middlewares are missing. One of them is [a middleware to add middlewares](https://github.com/nrepl/nrepl/blob/master/src/clojure/nrepl/middleware/dynamic_loader.clj) that Clojure Sublimed wants to use. I tried patching shadow-cljs, but no luck so far - I am stuck on [this error](https://github.com/nrepl/nrepl/blob/master/src/clojure/nrepl/middleware/dynamic_loader.clj#L69:L71).

I will post an update if I resolve this, but I wanted to share my findings anyway to save time for the next person trying to figure it out.

**Comment by tonsky** [comment]
Thanks @alekzvik! Another angle of attack might be for Clojure Sublimed to not require extra middlewares for CLJS

**Comment by alekzvik** [comment]
@tonsky, or they can be optional. A [describe op](https://nrepl.org/nrepl/ops.html#describe) can return a list of supported operations from the nREPL server. 
This approach might also be helpful in #29

**Comment by tonsky** [comment]
Should work in 2.8.0 via Connect shadow-cljs command
